python/whylogs/api/writer/whylabs.py

```diff
@@ -132,22 +155,40 @@ def _upload_whylabs(
                 "variable or on your write method"
             )
 
-        upload_url = upload_url or self._get_upload_url(dataset_timestamp=dataset_timestamp)
+        upload_url = upload_url or self._get_upload_url(
+            dataset_timestamp=dataset_timestamp, reference=reference, reference_alias=reference_alias
+        )
         api_key_id = self._api_key[:10] if self._api_key else None
-        try:
-            with open(profile_path, "rb") as f:
-                http_response = requests.put(upload_url, data=f.read())
-                if http_response.status_code == 200:
-                    logger.info(
-                        f"Done uploading {self._org_id}/{self._dataset_id}/{dataset_timestamp} to "
-                        f"{self.whylabs_api_endpoint} with API token ID: {api_key_id}"
-                    )
-                return http_response
-        except requests.RequestException as e:
-            logger.info(
-                f"Failed to upload {self._org_id}/{self._dataset_id}/{dataset_timestamp} to "
-                + f"{self.whylabs_api_endpoint} with API token ID: {api_key_id}. Error occurred: {e}"
-            )
+        if not reference:
+            try:
```

Comment:
I don't see a difference between the `if` and `else`branches?

Reply:
haha, yes. must be an old code I forgot to update.
fixed in the last commit